core/metacling/src/TCling.cxx

```diff
@@ -5867,6 +5877,25 @@ Int_t TCling::AutoParse(const char *cls)
    return nHheadersParsed > 0 ? 1 : 0;
 }
 
+// This is a function which gets callback from cling when DynamicLibraryManager->loadLibrary failed for some reason.
+// Try to solve the problem by autoloading. Return true when autoloading success, return
+// false if not.
+bool TCling::LibraryLoadingFailed(std::string errmessage, std::string libStem, bool permanent, bool resolved)
+{
+   StringRef errMsg(errmessage);
+   if (errMsg.contains("undefined symbol: ")) {
+      std::string mangled_name = std::string(errMsg.split("undefined symbol: ").second);
+      void* res = ((TCling*)gCling)->LazyFunctionCreatorAutoload(mangled_name);
```

Comment:
Do we have something similar for the (rarer) case of a global variables?

Reply:
This is for global variables.

Reply:
I am confused ... It mentions 'Function' in the method name ... and thus I would assume it would detect missing symbol for function but not for variables ... i.e.
```
   void ThisFunction();
```
vs
```
   int gGlobalVariable;
```

Reply:
It'll detect missing symbols for global variables, for example when
`TTree tree;` was a global variable and libTree.so was not loaded, I think we get an error "undefined symbol TTree".

Reply:
> It'll detect missing symbols for global variables, when
> TTree tree; was a global variable and libTree.so was not loaded, I think we get an error > "undefined symbol TTree".

This is not the case I am talking about.  it the case of two libraries, 
```
// libA.so
TTree gTree;
```
and
```
// libB.so
extern TTree gTree;
Long64_t somefunction() {
   return gTree.GetEntries();
}
```
Then
You load (explictly or implicitly) libTree.so
and the attempt to load libB.so

I am guessing that this is not supported yet (and was **not** supported before .. so really, I am just asking :) )

Reply:
I see, yeah I think it's not supported